cloud/pkg/common/informers/informers.go

```diff
@@ -0,0 +1,167 @@
+package informers
+
+import (
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/kubeedge/cloud/pkg/apis/devices/v1alpha2"
+	"github.com/kubeedge/kubeedge/cloud/pkg/apis/reliablesyncs/v1alpha1"
+	"github.com/kubeedge/kubeedge/cloud/pkg/common/client"
+)
+
+type newInformer func() cache.SharedIndexInformer
+
+type Informers interface {
+	EdgeSitePod(nodeName string) cache.SharedIndexInformer
+	Pod() cache.SharedIndexInformer
+	ConfigMap() cache.SharedIndexInformer
+	Secrets() cache.SharedIndexInformer
+	Service() cache.SharedIndexInformer
+	Endpoints() cache.SharedIndexInformer
+	Node() cache.SharedIndexInformer
+	EdgeNode() cache.SharedIndexInformer
+	ClusterObjectSync() cache.SharedIndexInformer
+	ObjectSync() cache.SharedIndexInformer
+	Device() cache.SharedIndexInformer
+	Start(stopCh <-chan struct{})
```

Comment:
If we put all core and crd informers in a global informers, maybe it's not friendly for the expansion and decoupling.

So I think we can use sharedinformer for the core apis and for the crd, just use themselves.

Reply:
done! PTAL :)

Reply:
please rebase until #2487 merge

Reply:
ok